drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```
@@ -0,0 +1,428 @@
+        # malware pid not found, analyze entire log instead
+        return None
+
+    # get all child processes of the malware process
+    malware_pids = get_all_child_processes(malware_process)
+
+    # return a list of all spawned malware processes (does not include process injection)
+    return malware_pids
+
+
+def get_drakvuf_feature_extractor(
+    calls: Iterator[Dict],
+) -> capa.features.extractors.drakvuf.extractor.DrakvufExtractor:
+    # wrapper routing for initializing the Drakvuf feature extractor
+    return capa.features.extractors.drakvuf.extractor.DrakvufExtractor.from_report(
+        calls
+    )
+
+
+def decode_json_lines(fd: BinaryIO) -> Iterator[Dict]:
+    # read and decode json lines into a list of dictionaries
+    for line in fd:
+        try:
+            # we use orjson for a small performance improvement
+            line_s = line.strip()
+            obj = orjson.loads(line_s)
+            yield obj
+        except (orjson.JSONDecodeError):
+            # sometimes Drakvuf reports bad method names and/or malformed JSON
+            logger.debug("bad drakvuf log line: %s", line)
+
+
+def dynamic_capa_analysis(
+    analysis_dir: Path,
+    rules: capa.rules.RuleSet,
+    malware_pids: Optional[List[int]] = None,
+) -> Tuple[Path, capa.render.result_document.MatchResults]:
+
+    # save all api calls and native calls into one list which gets sorted by capa later on.
+    calls = []
+
+    # read api calls
+    if (analysis_dir / "apimon.log").exists():
+        with (analysis_dir / "apimon.log").open("r", errors="ignore") as apimon_fd:
+            calls += list(decode_json_lines(apimon_fd))
+    else:
+        logger.debug("missing apimon.log file")
+
+    # read syscalls
+    if (analysis_dir / "syscall.log").exists():
+        with (analysis_dir / "syscall.log").open("r", errors="ignore") as syscall_fd:
+            calls += list(decode_json_lines(syscall_fd))
+    else:
+        logger.debug("missing syscall.log file")
+
```

nice to have - maybe raise an exception `if not calls` (so there's neither syscall.log nor apimon.log)